R/archive.R

```diff
@@ -145,14 +150,77 @@ epi_archive =
                     c("geo_type", "time_type"))) {
               Warn("`additional_metadata` names overlap with existing metadata fields \"geo_type\", \"time_type\".")
             }
-
+            
+            # Finish off with compactify
+            if (missing(compactify)) compactify = NULL
+            
             # Create the data table; if x was an un-keyed data.table itself,
             # then the call to as.data.table() will fail to set keys, so we
             # need to check this, then do it manually if needed
             key_vars = c("geo_value", "time_value", other_keys, "version")
             DT = as.data.table(x, key = key_vars)
             if (!identical(key_vars, key(DT))) setkeyv(DT, cols = key_vars)
-
+            
+            # functions for LOCF
+            # orders data frame to observe potential LOCF
+            order <- function(df) {
+              arrange(df,geo_value,version,time_value)
+            }
+            
+            # Check if previous entry is in group.
+            mutate_in_group <- function(df) {
+              mutate(df, in_group =
+                       replace_na(
+                         (geo_value == lag(geo_value) &
+                            version == lag(version)),
```

Comment:
- That's a separate issue, which I may account for later should I work on a new issue
- I overhauled this, so this issue should no longer be a problem
- I changed it to do an `all`/`any` with `filter` as to avoid this problem (`filter_all` also may be used from what I saw on the documentation)